zerver/actions/users.py

```diff
@@ -484,15 +484,12 @@ def do_deactivate_user(
     if settings.BILLING_ENABLED:
         from corporate.lib.stripe import RealmBillingSession
 
-    if _cascade:
-        # We need to deactivate bots before the target user, to ensure
-        # that a failure partway through this function cannot result
-        # in only the user being deactivated.
-        bot_profiles = get_active_bots_owned_by_user(user_profile)
-        for profile in bot_profiles:
-            do_deactivate_user(profile, _cascade=False, acting_user=acting_user)
-
     with transaction.atomic(savepoint=False):
+        if _cascade:
+            bot_profiles = get_active_bots_owned_by_user(user_profile)
+            for profile in bot_profiles:
+                do_deactivate_user(profile, _cascade=False, acting_user=acting_user)
```

Comment:
Not sure why this was not inside a transaction before, but should be fine inlcuding this inside the transaction.

Reply:
It is not important. I thought would be better to avoid deactivating just bot and not users in case of any failure. Will remove this commit once we decide how to move forward.